quantum/matrix.c

```diff
@@ -23,14 +23,43 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "quantum.h"
 
 #ifdef DIRECT_PINS
-static pin_t direct_pins[MATRIX_ROWS][MATRIX_COLS] = DIRECT_PINS;
-#elif (DIODE_DIRECTION == ROW2COL) || (DIODE_DIRECTION == COL2ROW)
+static const pin_t direct_pins[MATRIX_ROWS][MATRIX_COLS] = DIRECT_PINS;
+#else
+/* Let's default to EITHERWAY. This works regardless of COL2ROW or ROW2COL wiring. */
+#    if !defined(DIODE_DIRECTION)
+#        define DIODE_DIRECTION EITHERWAY
+#    endif
+
+/* Workaround for facew : MATRIX_COL_PINS has fewer number of elements than MATRIX_COLS */
+#    if defined(KEYBOARD_facew)
```

Comment:
This should be fixed in the keyboard code, not in core.

Reply:
Working on this in #8115.